### PR TITLE
Check runtime Maven version is aligned with wrapper Maven version

### PR DESCRIPTION
**Pull request by jprinet**
### Issue
When using the Maven daemon, the Maven version used can be different than the one specified by the Maven wrapper. The daemon is shipping a customized version of Maven, see [here](https://github.com/apache/maven-mvnd/issues/143) for details.
This can impact negatively the build cacheability as the Maven version is part of the cache key.

### Goal of the PR
The only way to fix this issue is to have the Maven daemon using a configurable Maven version, which was pushed back in the linked ticket.
The proposal here is to tag the builds encountering this issue and add a console log to warn the user.
This will allow to identify the impacted builds.

### CODE REVIEW

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (runningcode) [inline .mvn/gradle-enterprise-custom-user-data.groovy:101]: Regarding variable naming: instead of `current` since `current` could be used to refer to the maven daemon version or the wrapper version, can we use `runtimeMavenVersion`?

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (runningcode) [inline .mvn/gradle-enterprise-custom-user-data.groovy:108]: This will work for Quarkus but we if the maven wrapper is distributed from a custom URL will this still work?

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (runningcode) [inline .mvn/gradle-enterprise-custom-user-data.groovy:112]: would it be valuable to add a custom value with the wrapper version? I'm assuming that the runtime version is already part of the scan

**Review by runningcode — COMMENTED** [review]
Looks good, added some thoughts

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (jprinet) [reply .mvn/gradle-enterprise-custom-user-data.groovy:108]: I am parsing here a standard URL, this could not work with a custom URL indeed.
I am adding a comment to describe what is expected.

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (jprinet) [reply .mvn/gradle-enterprise-custom-user-data.groovy:112]: I didn't think it was necessary at first thought as the version can be obtained in the Git repo, let's add it to simplify investigation if any 👍

Comment (runningcode) [reply .mvn/gradle-enterprise-custom-user-data.groovy:112]: Good point. I'm not sure it is needed if it is in the repo then. It was just a thought.

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (runningcode) [inline .mvn/gradle-enterprise-custom-user-data.groovy:99]:
```suggestion
// Check current Maven version and Maven Wrapper version alignment
```

**Review by runningcode — APPROVED** [review]

File: .mvn/gradle-enterprise-custom-user-data.groovy
Comment (jprinet) [reply .mvn/gradle-enterprise-custom-user-data.groovy:114]: I'll add it, Guillaume will decide if they want it or not

**Review by gsmet — APPROVED** [review]
Looks good to me, will wait for at least one CI run to be green before merging.

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**
